### PR TITLE
Update to gRPC 1.15.0.

### LINKED ISSUE #18: Update to gRPC 1.15.0
Would it be possible to update this library to compile against gRPC 1.15.0 (or later)? 1.15.0 has moved to Java 7, so then it would be possible to use Java 7 features in this library (diamond operator, try-with-resources).

Also, I noticed that the generated protobuf code used in this project is compiled with a very old version of gRPC (0.15.0) that gives deprecation warnings when compiled with a newer gRPC library. Would it be possible to update the generated code to use a newer version of the protoc generated code or update the project to use protobuf-maven-plugin to generate code as part of the build?

I have a branch here with the suggested changes (https://github.com/opentracing-contrib/java-grpc/compare/master...pkwarren:update_grpc_1.15?expand=1) but wanted to run this by the maintainers first before raising a PR.

### PR DESCRIPTION
**Pull request by pkwarren**
Update to gRPC 1.15.0 (which enables this library to upgrade to Java 7).
Switch to use try-with-resources and diamond operator which are
available in Java 7. Remove deprecated use of `.usePlaintext(boolean)`.
Update protobuf code used by tests to be built with a more recent
version of the gRPC plug-in (1.15.0).

Fixes #18.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -228,6 +236,30 @@
...
+            io.grpc:protoc-gen-grpc-java:${grpc.version}:exe:${os.detected.classifier}
+          </pluginArtifact>
+        </configuration>
+        <executions>
+          <execution>
+            <id>proto-test-compile</id>
+            <phase>generate-test-sources</phase>
+            <goals>
+              <goal>test-compile</goal>
+              <goal>test-compile-custom</goal>
```
Comment (pkwarren) [inline pom.xml]: Switched from checking in generated code to generating it as part of the build.

File: src/test/java/io/opentracing/contrib/grpc/TracedClient.java [hunk 2]
```diff
@@ -25,7 +25,7 @@
 
   public TracedClient(String host, int port, ClientTracingInterceptor tracingInterceptor) {
     channel = ManagedChannelBuilder.forAddress(host, port)
-        .usePlaintext(true)
+        .usePlaintext()
```
Comment (pkwarren) [inline src/test/java/io/opentracing/contrib/grpc/TracedClient.java]: `usePlaintext(boolean)` is now deprecated.

File: src/test/proto/helloworld/helloworld.proto [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "io.opentracing.contrib.grpc.gen";
```
Comment (pkwarren) [inline src/test/proto/helloworld/helloworld.proto]: Taken from https://github.com/grpc/grpc-java/blob/ab5257504baf62e804f491748263bc9d760416b4/examples/src/main/proto/helloworld.proto. Only change is to `java_package` to reflect where generated code used to be checked in to source control.

**Comment by malafeev** [comment]
thanks @pkwarren

**Pull request merged**